src/vs/workbench/contrib/files/browser/views/explorerView.ts

```diff
@@ -507,27 +507,35 @@ export class ExplorerView extends ViewletPanel {
 		return toResource(input, { supportSideBySide: true });
 	}
 
-	private onSelectItem(fileStat: ExplorerItem, reveal = this.autoReveal): Promise<void> {
-		if (!fileStat || !this.isBodyVisible() || this.tree.getInput() === fileStat) {
-			return Promise.resolve(undefined);
+	private onSelectResource(resource: URI, reveal = this.autoReveal): void {
+		if (!resource || !this.isBodyVisible()) {
+			return;
 		}
 
 		// Expand all stats in the parent chain
-		const toExpand: ExplorerItem[] = [];
-		let parent = fileStat.parent;
-		while (parent) {
-			toExpand.push(parent);
-			parent = parent.parent;
-		}
-
-		return sequence(toExpand.reverse().map(s => () => this.tree.expand(s))).then(() => {
-			if (reveal) {
-				this.tree.reveal(fileStat, 0.5);
-			}
+		const root = this.explorerService.roots.filter(r => isEqualOrParent(resource, r.resource)).pop();
+		if (root) {
+			const traverse = async (item: ExplorerItem) => {
+				if (item.resource.toString() === resource.toString()) {
+					if (reveal) {
+						this.tree.reveal(item, 0.5);
+					}
 
-			this.tree.setFocus([fileStat]);
-			this.tree.setSelection([fileStat]);
-		});
+					this.tree.setFocus([item]);
+					this.tree.setSelection([item]);
+				} else {
+					await this.tree.expand(item);
+					let found = false;
+					item.children.forEach(async child => {
+						if (!found && isEqualOrParent(resource, child.resource)) {
```

Comment:
Sorry for the random comment, I was just kind of curious about this pattern here.

https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop
https://dev.to/burkeholland/asyncawait-and-the-foreach-pit-of-despair-2267
https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404

Researching online seems to indicate this may be a slight antipattern with recommendations to use Promises.all or just a simple for loop with await. Is that necessarily the case here?

Reply:
Here it does not matter, since there is only one promise which will be waited on I believe. So we don't even have to await it in practice.

Reply:
I thought the same once I saw it, ended up simplifying that code a lot.

Reply:
Thanks for simplifying!